smacdown-ensemble/script.whizzml

```diff
@@ -61,6 +61,12 @@
   (try (delete id)
        (catch e (log-info (str "Error deleting resource " id " ignored")))))
 
+(define (delete-batchprediction-datasets resources)
+  (let (ids (filter (lambda (id) (!= id ""))
+                    (map (lambda (r) ((fetch r) "output_dataset_resource" ""))
+                         resources)))
+    (map safe-delete ids)))
```

Comment:
since whizzml is not specially quick, maybe a single `map` would be
enough? something like:

```
(for (r resources)
  (let (id ((fetch r) "output_dataset_resource" false))
    (when id (safe-delete id))))
```

(i also think it's a bit easier to read)